packages/@css-blocks/ember-app/src/index.ts

```diff
@@ -182,14 +187,66 @@ const EMBER_ADDON: AddonImplementation<CSSBlocksApplicationAddon> = {
         throw new Error("[css-blocks/ember-app] The CSS tree ran before the JS tree, so the CSS tree doesn't have the contents for CSS Blocks files. This shouldn't ever happen, but if it does, please file an issue with us!");
       }
       // Get the combined CSS file
-      const cssBlocksContentsTree = new CSSBlocksStylesProcessorPlugin(env.modulePrefix, env.config, [this.broccoliAppPluginInstance, tree]);
+      const cssBlocksContentsTree = new CSSBlocksStylesPreprocessorPlugin(env.modulePrefix, env.config, [this.broccoliAppPluginInstance, tree]);
       return new BroccoliDebug(mergeTrees([tree, cssBlocksContentsTree], { overwrite: true }), "css-blocks:css-preprocess");
     } else {
       return tree;
     }
   },
+
+  postprocessTree(type, tree) {
+    let env = this.env!;
+
+    if (type === "css") {
+      if (!env.isApp || env.config.broccoliConcat === "SKIP") {
+        return tree;
+      }
+
+      // Merge default concat options with the options provided by the app.
+      let concatOptions: BroccoliConcatOptions;
+      if (env.config.broccoliConcat) {
+        concatOptions = Object.assign(
+          {},
+          buildDefaultBroccoliConcatOptions(env),
+          env.config.broccoliConcat,
+        );
+      } else {
+        concatOptions = buildDefaultBroccoliConcatOptions(env);
+      }
```

Comment:
It's slightly less efficient, but I think we should move the conditional check into the expression, it cleans up some of the duplication and makes this block easier to read.

```suggestion
      let concatOptions: BroccoliConcatOptions = Object.assign(
        {},
        buildDefaultBroccoliConcatOptions(env),
        env.config.broccoliConcat || {},
      );
```